src/geometry.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * Given a CSS px value (e.g. '10px'), strips the px suffix and returns the value as a number.
+ */
+function pxToNumber(value: string): number {
+	return Number(value.slice(0, -2));
+}
+
+/**
+ * Given a number, returns a CSS px value (e.g. '10px')
+ */
+function numberToPx(value: number): string {
+	return value + 'px';
+}
+
+/**
+ * Returns an object satisfying the ClientRect interface that is the result of adjusting the given ClientRect
+ * by the given dimensions.  The resulting object will have larger dimensions if the adjustment parameters
+ * are positive, or smaller dimensions if they are negative.
+ */
+function adjustBox(box: ClientRect, top: number, right: number, bottom: number, left: number): ClientRect {
+	return {
+		bottom: box.bottom + bottom,
+		height: box.height + top + bottom,
+		left: box.left - left,
+		right: box.right + right,
+		top: box.top - top,
+		width: box.width + left + right
+	};
+}
+
+export interface SettableClientRect {
+    height?: number;
```

Comment:
Was this copied directly from the proposal?  There are spaces instead of tabs here.

Also, as I learned some time after writing that proposal, an interface with all-optional keys ends up collapsing to `any` which isn't what we want here.  The alternative is a somewhat long-winded union type combining 4 objects (where one of the 4 keys is listed as required in each).